core/server/data/migrations/utils.js

```diff
@@ -443,33 +444,59 @@ function createDropColumnMigration(table, column, columnDefinition) {
 /**
  * @param {string} table
  * @param {string} column
- *
+ * @param {Object} options
+ * @param {boolean} options.disableForeignKeyChecks Disable foreign key checks for the down operation (when dropping nullable)
  * @returns {Migration}
  */
-function createSetNullableMigration(table, column) {
-    return createNonTransactionalMigration(
+function createSetNullableMigration(table, column, options = {}) {
+    return createTransactionalMigration(
         async function up(knex) {
             logging.info(`Setting nullable: ${table}.${column}`);
             await commands.setNullable(table, column, knex);
         },
         async function down(knex) {
-            logging.info(`Dropping nullable:  ${table}.${column}`);
+            if (DatabaseInfo.isSQLite(knex)) {
+                options.disableForeignKeyChecks = false;
+            }
+            logging.info(`Dropping nullable:  ${table}.${column}${options.disableForeignKeyChecks ? ' with foreign keys disabled' : ''}`);
+
+            if (options.disableForeignKeyChecks) {
+                await knex.raw('SET FOREIGN_KEY_CHECKS=0;').transacting(knex);
+            }
+
             await commands.dropNullable(table, column, knex);
+
+            if (options.disableForeignKeyChecks) {
+                await knex.raw('SET FOREIGN_KEY_CHECKS=1;').transacting(knex);
+            }
         }
     );
 }
 
 /**
  * @param {string} table
  * @param {string} column
- *
+ * @param {Object} options
+ * @param {boolean} options.disableForeignKeyChecks Disable foreign key checks for the up operation (when dropping nullable)
  * @returns {Migration}
  */
-function createDropNullableMigration(table, column) {
-    return createNonTransactionalMigration(
+function createDropNullableMigration(table, column, options = {}) {
+    return createTransactionalMigration(
         async function up(knex) {
-            logging.info(`Dropping nullable: ${table}.${column}`);
+            if (DatabaseInfo.isSQLite(knex)) {
+                options.disableForeignKeyChecks = false;
+            }
+            logging.info(`Dropping nullable: ${table}.${column}${options.disableForeignKeyChecks ? ' with foreign keys disabled' : ''}`);
+
+            if (options.disableForeignKeyChecks) {
+                await knex.raw('SET FOREIGN_KEY_CHECKS=0;').transacting(knex);
```

Comment:
Rolling back a transaction doesn't reset the session variable. IMO this should be in a try/finally instead of using a transaction.


<img width="320" alt="image" src="https://user-images.githubusercontent.com/3798302/166463730-4ae6533b-0685-4769-aa70-1725172f7539.png">